### ISSUE TITLE
Reason: authError, Message: Invalid Credentials while creating GKE NodePool

### ISSUE THREAD

**Issue by MaxenceAdnot** [terraform-providers/terraform-provider-google#6867]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

0.12.6

### Affected Resource(s)

google_container_node_pool

### Terraform Configuration Files

I'm making use of the safer-cluster-update-variant module, below you'll find the invocation of the module I'm using: 

```tf
module "nonprod-kubernetes" {
  source                  = "terraform-google-modules/kubernetes-engine/google//modules/safer-cluster-update-variant"
  version                 = "~> 10.0"

  project_id              = module.nonprod_enable_google_apis.project_id
  name                    = "automation-cluster"
  description             = "Automation Non Prod GKE Cluster"
  regional                = true
  region                  = var.default_region
  network                 = local.nonprod_host_network.name
  network_project_id      = google_project_iam_member.nonprod_compute_network_user_container.project
  subnetwork              = local.nonprod_shared_subnetwork.name
  ip_range_pods           = "automation-subnet-gke-pod"
  ip_range_services       = "automation-subnet-gke-svc"
  master_ipv4_cidr_block  = "172.16.200.0/28"
  registry_project_id     = data.google_projects.registry_service_project.projects[0].project_id
  enable_private_endpoint = true
  maintenance_start_time  = "00:00"
  master_authorized_networks = [{
    cidr_block   = "${module.nonprod_bastion.ip_address}/32"
    display_name = "Bastion Host"
  }]
  grant_registry_access = true
  node_pools = [
    {
      name         = "standard-pool"
      min_count    = 1
      max_count    = 2
      machine_type = "n1-standard-2"
      auto_upgrade = true
    }
  ]
  node_pools_tags = {
    all = ["allow-lb", "egress-internet"]
  }
  default_max_pods_per_node = 100
}
```

### Debug Output

Cannot provide it yet

### Expected Behavior

I expected the terraform apply to finish properly.

### Actual Behavior

Terraform apply is giving me this output after a while: 

```
module.nonprod-kubernetes.module.gke.google_container_node_pool.pools["standard-pool"]: Still creating... [3m20s elapsed]

Error: Error waiting for creating GKE NodePool: error while retrieving operation: googleapi: Error 401: Request had invalid authentication credentials. Expected OAuth 2 access token, login cookie or other valid authentication credential. See https://developers.google.com/identity/sign-in/web/devconsole-project.
More details:
Reason: authError, Message: Invalid Credentials


  on .terraform/modules/nonprod-kubernetes/terraform-google-kubernetes-engine-10.0.0/modules/beta-private-cluster-update-variant/cluster.tf line 341, in resource "google_container_node_pool" "pools":
 341: resource "google_container_node_pool" "pools" {
```

### Steps to Reproduce

Basically just 
1. `terraform apply`

However please note that I managed a couple of time to get it working (by chance I guess) without any issues.

### Important Factoids

I integrated the module for GKE in my copy of Terraform Cloud Foundation Toolkit (https://github.com/terraform-google-modules/terraform-example-foundation/), Terraform is using the SA created by in the step 0-bootstrap: 
https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/3-projects/providers.tf

* #0000

**Comment by edwardmedia** [comment]
@MaxenceAdnot I can't run your code as key configs are in hidden modules. To verify if there are issues with `GKE NodePool`, can you run examples from below links?
https://www.terraform.io/docs/providers/google/r/container_node_pool.html

**Comment by MaxenceAdnot** [comment]
I refactored my module and managed to get it working 100% of the time. I also tested the GKE NodePool on its own with no context and it worked too. I guess the problem was between the chair and the keyboard... 

Sorry for the inconvenience then and thank you for your support.
